scripts/add-cluster.sh

```diff
@@ -99,13 +109,13 @@ SA_TOKEN=`oc get secret ${SA_SECRET} -n ${OPERATOR_NS}  -o json | jq -r '.data["
 SA_CA_CRT=`oc get secret ${SA_SECRET} -n ${OPERATOR_NS} -o json | jq -r '.data["ca.crt"]'`
 
 # this env variable is set in openshift-ci environment.
-# openshift ci has long name for cluster i.e.> 63 characters if read from config which is not allowed by k8s/openshift
-if [[ -z ${OPENSHIFT_BUILD_NAMESPACE} ]]; then
+# openshift 4 has long name for cluster i.e.> 63 characters if read from config which is not allowed by k8s/openshift
+if [[ ${MINISHIFT} == "true" ]]; then
```

Comment:
I think it's breaks the local deployments when using `make test-e2e` and `make test-e2e-local` from e2e repo when running against minishift. I don't think we set minishift flag in this case.
We would need to use something to automatically detect the version of OS here instead of relaying on the flag.

Reply:
we should be able to easily check if the OC is version 3 or 4: 
```
IS_OS3=`curl -k -XGET -H "Authorization: Bearer $(oc whoami -t 2>/dev/null)" $(oc config view --minify -o jsonpath='{.clusters[0].cluster.server}')/version/openshift 2>/dev/null | grep paths`
```
if the OpenShift is of version 4, then the variable is empty (in OS4 the `/version/openshift` endpoint is missing), if it is of version 3 (in other words is minishift) then the var is not empty

Reply:
in other words - we could easily remove the parameter `--minishift` and replace it by:
```
MINISHIFT=`curl -k -XGET -H "Authorization: Bearer $(oc whoami -t 2>/dev/null)" $(oc config view --minify -o jsonpath='{.clusters[0].cluster.server}')/version/openshift 2>/dev/null | grep paths`
```

Reply:
Btw, why do we need to change the `if` check? For now, it was working fine with the 48h clusters (that are OS4), so what is the difference in the case of dev clusters?

Reply:
> Btw, why do we need to change the if check? For now, it was working fine with the 48h clusters (that are OS4), so what is the difference in the case of dev clusters?

With the current logic, we are using `OPENSHIFT_BUILD_NAMESPACE`  env variable, which is set only in openshift-ci environment. If we are running 48 hrs dev cluster, we don't have `OPENSHIFT_BUILD_NAMESPACE`  variable set and I don't want that to be set explicitly by user even not needed

Reply:
I understand the situation with `OPENSHIFT_BUILD_NAMESPACE` env variable and I didn't ask you about it. 
The reason why we have different ways of getting  `API_ENDPOINT` and `JOINING_CLUSTER_NAME` vas is (from the comment in the script):
`# openshift ci has long name for cluster i.e.> 63 characters if read from config which is not allowed by k8s/openshift`
which is the case for openshift-ci, but not for dev clusters. In other words - why we cannot take the values from the kubeconfig also for the dev clusters?

Anyway, it's fine if we change the if statement to use kubeconfig only for OS3 minishift, but use the curl command that I posted above, so we don't have to change the toolchain-e2e test setup. I was just curious why do we need to do that - if there is another reason...

Reply:
it was only reason which is added in the comment, but as we are getting cluster name from infrastucture CRD for openshift 4, let's use it for openshift 4 to be consistent.

Also it won't change if we read it from config for dev clusters or openshift-ci and both are creating new openshift 4 cluster.

Reply:
Addressed in d4f3d0d